sparsecoding/data/datasets/datasets.py

```diff
@@ -90,46 +87,45 @@ def whiten(data):
     wdata - scalar (N,n)
         whitened data
     """
-    
+
     cdata = data.T-data.mean()
-    
+
     cov = np.cov(cdata)
     w, v = np.linalg.eig(cov)
     diagw = np.diag(np.real(1/(w**0.5)))
-    
+
     wdata = v@diagw@v.T@cdata
     return wdata.T.astype(np.float32)
 
 
-def whiten_dataset(dataset,plot=False,title='',n_comp=-1):
+def whiten_dataset(dataset, plot=False, title='', n_comp=-1):
     from torchsmodel import sparsecoding
     from utils import plotmontage
-    
+
     # zero mean
     cpatches = (torchtonp(dataset.patches)-torchtonp(dataset.patches).mean(axis=0)).T
     # covariance
     cov = np.cov(cpatches)
     # eigendecomp.
     w, v = np.linalg.eig(cov)
     diagw = np.diag(np.real(1/(w**0.5)))
-    
+
     # plot componenets
     if plot:
         model = sparsecoding(n_basis=(3*dataset.patch_size)**2,
                              n=(3*dataset.patch_size)**2)
 
         model.D = torch.from_numpy(v).to(torch.device("cpu"))
-        plotmontage(model,color=True,size=10,title=title)
-
-    # whiten 
-    trotpatch = (nptotorch(v,device=dataset.device).t()@dataset.patches.t()).t()
-    twpatch = nptotorch(np.zeros(trotpatch.shape),device=dataset.device)
-    twpatch[:,:n_comp] = trotpatch[:,:n_comp]
-    dataset.patches = (nptotorch(v,device=dataset.device)@nptotorch(diagw,device=dataset.device)@twpatch.t()).t()
+        plotmontage(model, color=True, size=10, title=title)
 
+    # whiten
+    trotpatch = (nptotorch(v, device=dataset.device).t()@dataset.patches.t()).t()
+    twpatch = nptotorch(np.zeros(trotpatch.shape), device=dataset.device)
+    twpatch[:, :n_comp] = trotpatch[:, :n_comp]
+    dataset.patches = (nptotorch(v, device=dataset.device)@nptotorch(diagw, device=dataset.device)@twpatch.t()).t()
 
 
-def nptotorch(x,device=torch.device('cpu'),dtype=np.float32):
+def nptotorch(x, device=torch.device('cpu'), dtype=np.float32):
```

Comment:
nptotorch -> np_to_torch

Reply:
Do we need this function? 
You can also say
`torch.tensor(v, device=device, dtype=dtype)`

See torch.torch
https://pytorch.org/docs/stable/generated/torch.tensor.html